spinalcordtoolbox/reports/slice.py

```diff
@@ -286,9 +293,16 @@ def _resample_slicewise(self, image, p_resample, type_img, image_ref=None):
         nii = Nifti1Image(image.data, image.hdr.get_best_affine())
         # If no reference image is provided, resample to specified resolution
         if image_ref is None:
-            # Resample to px x p_resample x p_resample mm (orientation is SAL by convention in QC module)
-            nii_r = resample_nib(nii, new_size=[image.dim[4], p_resample, p_resample], new_size_type='mm',
-                                 interpolation=dict_interp[type_img])
+            slice_type = self.get_name()
+            if slice_type == 'Axial':
```

Comment:
Note: The `.get_name()` method is removed in #4126.

So, we should either a) keep it in #4126, b) find another way to get the slice type, or c) make these class methods instead (which seems like a lot of duplicated code) to me.

I'm leaning towards a) because it requires the fewest changes, but if there was a way to do c) without duplicating too much code, it seems like the most... Pythonic? (Though, it would require a greater refactor.)

Reply:
Ah, wait, I have an idea!

Reply:
I've tried thinking of a way to make 3. work. For example, we could create a method in each slice called `get_slice_dim`, and it returns the dimensions used for slices in a given SAL image. (e.g. for Axial it returns `[1, 2]` (AP, LR) and for Sagittal it returns `[0, 1]` (SI, AP). Then, those values are used to determine `new_size`, and the resampling call stays the same.

But, this just felt a bit too convoluted. Plus, `Slice` objects already have some suspect Sagittal and Axial-specific methods anyway. So, I think we can save Pythonic inheritance patterns for a separate refactoring PR of `slice.py`.

Reply:
#4126 has been merged, and `get_name()` has been kept for now, meaning that we're going for 1. at the moment.

Reply:
> Plus, `Slice` objects already have some suspect Sagittal and Axial-specific methods anyway (e.g. `get_sagittal_dim` and `get_axial_dim`? why not just `get_dim` as an abstract method?). So, I think we can save aiming for Pythonic inheritance patterns for a separate refactoring PR of `slice.py`.

That sounds like a good idea, yes.

I'll point out another, slightly more Pythonic (but still not great) way to do Option 2: testing the slice object with `isinstance(self, Axial)`, rather than doing a string comparison which boils down to `type(self).__name__ == 'Axial'`.